src/lib/files.ts

```diff
@@ -70,6 +73,41 @@ export class JSONWriter {
     const revoke = () => window.URL.revokeObjectURL(url);
     return [url, revoke];
   };
+
+  download = (): void => {
+    const [fileURL, revokeURL] = this.toURL();
+
+    new FileUI().download(
+      `qboard-${FileUI.timeStringNow()}.json`,
+      fileURL,
+      revokeURL
+    );
+  };
```

Comment:
Maybe this should actually return an object of a new type `FileUIDownloader` with the data
```js
{
  name: `qboard-${FileUI.timeStringNow()}.json`,
  fileURL,
  revokeURL,
}
```
Then `FileUI::download` should accept an object of this type, and any function calling `JSONWriter::download` should take the result and call its own `FileUI::download`. I'd probably rename `JSONWriter::download` to `JSONWriter::getDownloader`.

(I use the `::` syntax to denote instance methods.)